xmrstak/misc/executor.cpp

```diff
@@ -524,32 +531,38 @@ void executor::ex_main()
 		if(!cfg.tls)
 			dev_tls = false;
 
-		if(!xmrstak::params::inst().poolURL.empty() && xmrstak::params::inst().poolURL == cfg.sPoolAddr)
+		auto& pool_from_params = find_if(xmrstak::params::inst().pools.begin(), xmrstak::params::inst().pools.end(), [&cfg](const xmrstak::pool &pool_from_params) { return pool_from_params.poolURL == cfg.sPoolAddr; });
```

Comment:
This line produces an error in linux: 

```
xmrstak/misc/executor.cpp:534:35: error: invalid initialization of non-const reference of type ‘__gnu_cxx::__normal_iterator<xmrstak::pool*, std::vector<xmrstak::pool> >&’ from an rvalue of type ‘__gnu_cxx::__normal_iterator<xmrstak::pool*, std::vector<xmrstak::pool> >’
   auto& pool_from_params = find_if(xmrstak::params::inst().pools.begin(), xmrstak::params::inst().pools.end(), [&cfg](const xmrstak::pool &pool_from_params) { return pool_from_params.poolURL == cfg.sPoolAddr; });
                            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
When using "auto" as variable type, compiler doesn't automatically use "const" qualifier, so in this case it should be "const auto&" for the line to be valid. It is always best to explicitly add "const" qualifier if the following code is not supposed to modify the variable contents, this will allow compiler to optimize the code better.